File: LeetCode/1464. Maximum Product of Two Elements in an Array/solution.cpp
/**
 * 104 / 104 test cases passed.
 * Status: Accepted
 * Runtime: 12 ms
 */
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int first = 1;
        int second = 0;
        if (nums[first] < nums[second]) swap(first, second);
        for (int i = 2; i < nums.size(); ++ i) {
            if (nums[i] > nums[first]) {
                second = first;
                first = i;
            } else if (nums[i] > nums[second]) {
                second = i;
            }
        }
        return (nums[first] - 1) * (nums[second] - 1);
    }
};

File: LeetCode/1464. Maximum Product of Two Elements in an Array/solution.py
"""
104 / 104 test cases passed.
Status: Accepted
Runtime: 40 ms
"""
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        sorted_nums = sorted(nums)
        return (sorted_nums[-1] - 1) * (sorted_nums[-2] - 1)

"""
104 / 104 test cases passed.
Status: Accepted
Runtime: 44 ms
"""
class Solution2:
    def maxProduct(self, nums: List[int]) -> int:
        maximun = max(nums)
        nums.remove(maximun)
        return (maximun - 1) * (max(nums) - 1)
